### PR TITLE
Minor refactors to processing module

### PR DESCRIPTION
**Pull request by adarshsanjeev**
Refactors a few things.

- Adds SemanticUtils maps to columns.
- Add some addAll functions to reduce duplication, and for future reuse.
- Refactor `VariantColumnAndIndexSupplier` to only take a `SmooshedFileMapper` instead.
- Refactor `LongColumnSerializerV2` to have separate functions for serializing a value and null.

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/segment/LongColumnSerializerV2.java [hunk 1]
```diff
@@ -121,12 +121,23 @@ public void open() throws IOException
...
-      writer.add(0L);
+      serializeNull();
     } else {
-      writer.add(selector.getLong());
+      serializeValue(selector.getLong());
     }
-    rowCount++;
+  }
+
+  public void serializeNull() throws IOException
```
Comment (cryptoe) [inline processing/src/main/java/org/apache/druid/segment/LongColumnSerializerV2.java:130]: Please add java docs for these public methods.

File: processing/src/main/java/org/apache/druid/segment/nested/NestedDataColumnSerializer.java [hunk 2]
```diff
@@ -374,11 +375,17 @@ public void serializeDictionaries(
...
+
+  public void serialize(StructuredData data) throws IOException
   {
     if (!dictionarySerialized) {
-      throw new ISE("Must serialize value dictionaries before serializing values for column [%s]", name);
+      throw DruidException.defensive(
+          "Must serialize value dictionaries before serializing values for column [%s]", name
+      );
     }
-    StructuredData data = StructuredData.wrap(selector.getObject());
```
Comment (cryptoe) [inline processing/src/main/java/org/apache/druid/segment/nested/NestedDataColumnSerializer.java:381]: The order is changed a bit. 
Earlier, exception was checked first and then the wrapping was done. Is that a problem ?

**Review by cryptoe — APPROVED** [review]

File: processing/src/main/java/org/apache/druid/segment/LongColumnSerializerV2.java (hunk 1 above)
Comment (adarshsanjeev) [reply processing/src/main/java/org/apache/druid/segment/LongColumnSerializerV2.java:133]: Added

File: processing/src/main/java/org/apache/druid/segment/nested/NestedDataColumnSerializer.java (hunk 2 above)
Comment (adarshsanjeev) [reply processing/src/main/java/org/apache/druid/segment/nested/NestedDataColumnSerializer.java:390]: StructuredData.wrap is a simple constructor, so I think this is fine

**Pull request merged**